packages/datadog-plugin-aws-sdk/test/index.spec.js

```diff
@@ -0,0 +1,1137 @@
+'use strict'
```

Comment:
Is there any way to not repeat the main tests for every service? I wonder if maybe a few of the it blocks could be extracted to a reusable function, at least for any generic tags/propagation/etc.

Reply:
Mocha is fighting me to the death on this, so trying to work on this still, the before/after setup/teardown does not want to play nice with abstracting some of these tests into a reusable function.

Should hopefully have something here by EOD

Reply:
So i've broken out there tests into as reusable of functions as I could, and ensured that i'm now properly testing each generic thing for each service, the drawback here is that these tests now take a very long time, they will probably time out on the ci, so i need to work on getting them sped up somehow

Reply:
Nice, I think you actually pushed it even further than what I had in mind at that point 😄